fdbserver/workloads/BlobRestoreWorkload.actor.cpp

```diff
@@ -106,40 +107,92 @@ struct BlobRestoreWorkload : TestWorkload {
 					return Void();
 				}
 			}
-			wait(delay(1));
+			wait(delay(5));
 		}
 	}
 
-	ACTOR static Future<Void> copyToOriginalDb(Database cx, BlobRestoreWorkload* self) {
-		state RangeResult data;
-
-		// Read data from restored db
-		state Transaction tr1(self->extraDb_->clone());
+	ACTOR static Future<Standalone<VectorRef<KeyValueRef>>> read(Database cx, BlobRestoreWorkload* self) {
+		state Standalone<VectorRef<KeyValueRef>> data;
+		state Transaction tr(cx);
+		state GetRangeLimits limits(self->rwBatchSize_);
+		limits.minRows = 0;
+		state KeySelectorRef begin = firstGreaterOrEqual(normalKeys.begin);
+		state KeySelectorRef end = firstGreaterOrEqual(normalKeys.end);
+		state Arena arena;
 		loop {
 			try {
-				RangeResult result = wait(tr1.getRange(normalKeys, CLIENT_KNOBS->TOO_MANY));
-				data = result;
-				break;
+				Version ver = wait(tr.getReadVersion());
+				fmt::print("Read version {} \n", ver);
+				RangeResult result = wait(tr.getRange(begin, end, limits, Snapshot::True));
+				for (auto& row : result) {
+					data.push_back_deep(data.arena(), KeyValueRef(row.key, row.value));
+				}
+				if (!result.more) {
+					break;
+				}
+				if (result.readThrough.present()) {
+					begin = firstGreaterOrEqual(KeyRef(arena, result.readThrough.get()));
+				} else {
+					begin = firstGreaterThan(KeyRef(arena, result.end()[-1].key));
```

Comment:
Nit: result.back() is slightly cleaner than result.end()[-1]

Reply:
fixed. thanks